dask/dataframe/core.py

```diff
@@ -63,22 +84,47 @@ def finalize(results):
 
 
 class Scalar(Base):
-    """ A Dask-thing to represent a scalar
-
-    TODO: Clean up this abstraction
-    """
+    """ A Dask object to represent a pandas scalar"""
 
     _optimize = staticmethod(optimize)
     _default_get = staticmethod(threaded.get)
     _finalize = staticmethod(first)
 
-    def __init__(self, dsk, _name, name=None, divisions=None):
+    def __init__(self, dsk, name, meta, divisions=None):
+        # divisions is ignored, only present to be compatible with other
+        # objects.
         self.dask = dsk
-        self._name = _name
-        self.divisions = [None, None]
+        self._name = name
+        self._meta = make_meta(meta)
+
+    @property
+    def _meta_nonempty(self):
+        return self._meta
+
+    def __getattr__(self, key):
+        if key == 'dtype':
+            return self._meta.dtype
```

Comment:
If it was a property, it would appear even if it's not available. Slightly hacky, but works fine (we do the same for `cat` on `Series`). If more of these get added, I'd want to do something different.